examples/bevy/main.rs

```diff
@@ -0,0 +1,337 @@
+    out property <float> selected-green <=> green.value;
+    out property <float> selected-blue <=> blue.value;
+
+    in property <bool> show-loading-screen: true;
+    in property <string> download-url;
+    in property <percent> download-progress;
+
+    title: @tr("Slint & Bevy");
+    preferred-width: 500px;
+    preferred-height: 600px;
+
+    VerticalLayout {
+        alignment: start;
+        GroupBox {
+            title: "Light Color Controls";
+
+            GridBox {
+                Row {
+                    Text {
+                        text: "Red:";
+                        vertical-alignment: center;
+                    }
+
+                    red := Slider {
+                        minimum: 0.1;
+                        maximum: 1.0;
+                        value: 0.2;
+                    }
+                }
+
+                Row {
+                    Text {
+                        text: "Green:";
+                        vertical-alignment: center;
+                    }
+
+                    green := Slider {
+                        minimum: 0.1;
+                        maximum: 1.0;
+                        value: 0.5;
+                    }
+                }
+
+                Row {
+                    Text {
+                        text: "Blue:";
+                        vertical-alignment: center;
+                    }
+
+                    blue := Slider {
+                        minimum: 0.1;
+                        maximum: 1.0;
+                        value: 0.9;
+                    }
+                }
+            }
+        }
+
+        i := Image {
+            image-fit: fill;
+            width: 100%;
+            height: 100%;
+            preferred-width: self.source.width * 1px;
+            preferred-height: self.source.height * 1px;
+
+            if show-loading-screen: Rectangle {
+                background: Palette.background;
+                VerticalBox {
+                    alignment: start;
+                    Text {
+                        horizontal-alignment: center;
+                        text: "Downloading Assets";
+                    }
+                    Text {
+                        text: download-url;
+                        overflow: elide;
+                    }
+                    ProgressIndicator {
+                        indeterminate: download-url.is-empty;
+                        progress: root.download-progress;
+                    }
+                }
+            }
+        }
+    }
+}
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let (download_progress_sender, download_progress_receiver) =
+        smol::channel::bounded::<(SharedString, f32)>(5);
+
+    let (light_color_sender, light_color_receiver) =
+        smol::channel::bounded::<bevy::color::LinearRgba>(1);
+
+    let (new_texture_receiver, control_message_sender) =
+        spin_on::spin_on(slint_bevy_adapter::run_bevy_app_with_slint(
```

Comment:
🙈
I guess we can't use slint::spawn_local for this because it needs to happen before we initialize the event loop?  
This is a bit annoying.
Why is it async?

Reply:
Exactly, we need to intialize those early. It's asynchronous because multiple wgpu setup calls are async. They're know to be ready on first poll on all platforms except wasm.